custom_scripts/py_scripts/db_scanner/db_scanner.py

```diff
@@ -0,0 +1,255 @@
+    """
+    Load hosts from the file (useful when it's separated with Grinder)
+    :param filename: filename of file with hosts
+    :return: list of IP-addresses (for this case port is defined)
+    """
+    ip_list = []
+    with open(filename, mode="r") as json_hosts:
+        result_list = load(json_hosts)
+    for value in result_list:
+        if "ip" in value:
+            ip_list.append(value["ip"])
+    return ip_list
+
+
+def save_results(
+    host_dictionary: dict,
+    filename: str = DBScannerDefaultValues.RESULTS_DIRECTORY.joinpath(
+        DBScannerDefaultValues.JSON_RESULTS
+    ),
+) -> None:
+    """
+    Save results from execution to assigned file
+    :param host_dictionary: scanning results
+    :param filename: filename of resulting file
+    :return: None
+    """
+    host_result = dumps(host_dictionary, indent=4)
+    with open(filename, mode="a") as fd:
+        fd.write(host_result + "\n")
+
+
+def find_values_by_key(document: dict or list) -> dict or list or None:
+    """
+    Find different possible critical information by set regular expressions
+    :param document: document from the database
+    :return: found possible critical data
+    """
+    # print(type(document), document)
+    document_list = {}
+    for d in document:
+        # print(f"doc is {d} and type is {type(d)} of value's {type(document[d])}")
+        if not isinstance(document[d], list and dict):
+            for info in DBScannerDefaultValues.CRITICAL_INFORMATION:
+                # print(info, d)
+                regular = re.findall(f"{info}", d)
+                if len(regular) > 0:
+                    if isinstance(document[d], int or float):
+                        if document[d] != 0:
+                            document_list.update({f"{d}": document[d]})
+                    if document[d] not in ["", None, [], {}]:
+                        document_list.update({f"{d}": document[d]})
+                    break
+        else:
+            if isinstance(document[d], list):
+                for item in document[d]:
+                    # print(f"item is: {item}, type is: {type(item)}")
+                    if type(item) == list:
+                        for value in item:
+                            if isinstance(value, dict):
+                                result_list = find_values_by_key(value)
+                                # print(f"result list: {result_list}")
+                                if result_list is not None:
+                                    document_list.update({value: result_list})
+                    if isinstance(item, dict):
+                        result_list = find_values_by_key(item)
+                        # print(f"result list: {result_list}")
+                        if result_list is not None:
+                            document_list.update({item: result_list})
+            if isinstance(document[d], dict):
+                result_list = find_values_by_key(document[d])
```

Comment:
Nesting is huge. We totally need to separate this code into subfunction with comments, documentation etc.